### PR TITLE
Fixes #6916 by making it impossible to fuck up AME construction by making AME construction not awful

### LINKED ISSUE #6916: Anti-matter engine bug
(WEB REPORT BY: bradstein REMOTE: 158.69.55.2:7777)

> Revision (Should be above if you're viewing this from ingame!)

http://i.imgur.com/1bUsaon.png

> General description of the issue

Setting up Anti-matter engine and the final 3 compact modules didn't fully "unpack" even though they were unpacked properly with adjacent modules. 

> What you expected to happen

Make the antimatter engine with two cores and 12 completed pieces.

> What actually happened

Anti-matter engine with one core and 9 parts, other 3 wouldn't fully turn into the final pieces. 

> Steps to reproduce if possible

Do normal steps to making and securing antimatter engine.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by JustSumBody** [comment]
>making it impossible to fuck up AME construction

please
how will I meme people that do things like this
dreamseeker_2016-10-26_05-37-08
dreamseeker_2016-09-18_13-46-46

**Comment by Shadowmech88** [comment]
I also think AME construction should remain fuck-upable in some way.
It's already the safest engine, no need to make it any more braindead than it already is to set up.

**Comment by 9600bauds** [comment]
It can still fail to make cores if you build it in retard shapes, this just makes it so that you can't build unconnected pieces that would bug out or delete themselves. Especially if you build it too fast.

Note that the updateiconcode of this thing can still make invisible tiles under certain retard shapes.

**Comment by Shadowmech88** [comment]
Oh, that sounds fine then.

File: code/modules/power/antimatter/shielding.dm [hunk 1]
```diff
@@ -30,13 +31,34 @@ proc/cardinalrange(var/center)
...
-		controllerscan()
 
+/obj/machinery/am_shielding/proc/link_control(var/obj/machinery/power/am_control_unit/AMC)
+	if(!istype(AMC))
+		return 0
+	if(control_unit && control_unit != AMC)
+		return 0//Already have one
+	control_unit = AMC
+	return control_unit.add_shielding(src,1)
+	return 1
```
Comment (ComicIronic) [inline code/modules/power/antimatter/shielding.dm]: Extra return.

Comment (9600bauds) [reply code/modules/power/antimatter/shielding.dm]: How magical

**Pull request merged**